test/ServerSpec.js

```diff
@@ -65,4 +67,115 @@ describe('Server Spec', () => {
         .end(done);
     });
   });
+
+  describe('/search/:market/:checkin/:checkout', () => {
+    it('Should retrieve all San Francisco listings available for checkin 2017-11-12 checkout 2017-11-13', (done) => {
+      request(server)
+        .get('/search/San%20Francisco/2017-11-12/2017-11-13')
+        .expect('Content-Type', /json/)
+        .expect(200)
+        .expect((response) => {
+          const availableListings = response.body.filter(result => result.inventory_date.split('T')[0] === '2017-11-12');
+          expect(availableListings.length).to.equal(response.body.length);
+        })
+        .end(done);
+    });
+
+    it('Should retrieve no San Francisco listings available for checkin 2017-10-02 checkout 2017-10-03', (done) => {
+      request(server)
+        .get('/search/San%20Francisco/2017-10-02/2017-10-03')
+        .expect('Content-Type', /json/)
+        .expect(200)
+        .expect((response) => {
+          expect(response.body.length).to.equal(0);
+        })
+        .end(done);
+    });
+
+    it('Should retrieve all San Francisco listings available for checkin 2017-10-19 and checkout 2017-10-24', (done) => {
+      const availableListings = new Set();
+      const stayDates = {
+        '2017-10-19': 0, '2017-10-20': 0, '2017-10-21': 0, '2017-10-22': 0, '2017-10-23': 0,
+      };
+      request(server)
+        .get('/search/San%20Francisco/2017-10-19/2017-10-24')
+        .expect('Content-Type', /json/)
+        .expect(200)
+        .expect((response) => {
+          response.body.forEach((result) => {
+            availableListings.add(result.listing_id);
+            stayDates[result.inventory_date.split('T')[0]] += 1;
+          });
+          Object.keys(stayDates).forEach((date) => {
+            expect(stayDates[date]).to.equal(availableListings.size);
+          });
+        })
+        .end(done);
+    });
+  });
+
+  describe('/search/:market/:checkin/:checkout/:limit', () => {
+    it('Should retrieve 2 of 4 San Francisco listings available for checkin 2017-11-12 checkout 2017-11-13', (done) => {
```

Comment:
Isn't this test redundant to "Should retrieve 1 of 2 San Francisco listings available for checkin 2017-11-10 checkout 2017-11-13"?

Reply:
This one checks a one day stay; the 1 or 2 one checks a stay that spans multiple days.